spec/lib/manageiq/api/common/headers_spec.rb

```diff
@@ -1,13 +1,36 @@
 describe ManageIQ::API::Common::Headers do
-  let(:header_good) { ActionDispatch::Http::Headers.new({:blah => 'blah'}) }
-  let(:header_bad)  { {:blah => 'blah' } }
+  let(:header_good) do
+    ActionDispatch::Http::Headers.from_hash(:blah           => 'blah',
+                                            'x-rh-identity' => 'abc')
+  end
+  let(:header_bad) { {:blah => 'blah' } }
+
   context "set current headers" do
     it 'sets if the class is correct' do
-      expect(ManageIQ::API::Common::Headers.current = header_good).to be_a(ActionDispatch::Http::Headers)
+      described_class.with_headers(header_good) do |instance|
+        expect(instance).to be_a(ActionDispatch::Http::Headers)
+      end
     end
 
     it 'raises an exception if the class is incorrect' do
-      expect { ManageIQ::API::Common::Headers.current = header_bad }.to raise_exception(ArgumentError)
+      expect do
+        described_class.with_headers(header_bad) {}
+      end.to raise_exception(ArgumentError)
+    end
+  end
+
+  context "get forwardable headers" do
+    it "x-rh-identity" do
+      described_class.with_headers(header_good) do |_x|
```

Comment:
Minor, but the block var is not needed at all...that is you can just do `do` without the `|_x|`